src/VirtoCommerce.MarketingModule.Web/Controllers/Api/MarketingModulePromotionController.cs

```diff
@@ -90,10 +89,11 @@ public async Task<ActionResult<PromotionSearchResult>> PromotionsSearch([FromBod
         /// <param name="context">Promotion evaluation context</param>
         [HttpPost]
         [Route("evaluate")]
-        public async Task<ActionResult<PromotionReward[]>> EvaluatePromotions([FromBody]PromotionEvaluationContext context)
+        public async Task<ActionResult<webModel.PromotionReward[]>> EvaluatePromotions([FromBody]PromotionEvaluationContext context)
         {
-            var retVal = await _promoEvaluator.EvaluatePromotionAsync(context);
-            return Ok(retVal.Rewards);
+            var promotionResult = await _promoEvaluator.EvaluatePromotionAsync(context);
+            var result = promotionResult.Rewards.Select(x => AbstractTypeFactory<webModel.PromotionReward>.TryCreateInstance().FromModel(x));
```

Comment:
Could you do this  `Duck typing` conversion from `core.PromotionReward`  to `web.PromotionReward`  thought JSON serialization or by another way? This allows us to eliminate the explicit implementation of the `PromotionReward.FromModel` method.